gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dsubmenulist_ttt2.lua

```diff
@@ -0,0 +1,217 @@
+---
+-- @class PANEL
+-- @section DSubmenuListTTT2
+
+local PANEL = {}
+
+---
+-- @accessor function
+-- @realm client
+AccessorFunc(PANEL, "searchFunction", "SearchFunction")
+
+-- Define sizes
+local heightNavHeader = 10
+local heightNavButton = 50
+
+---
+-- Checks recursively the parents until none is found and the highest parent is returned
+-- @ignore
+local function getHighestParent(slf)
+	local parent = slf
+	local checkParent = slf:GetParent()
+
+	while ispanel(checkParent) do
+		parent = checkParent
+		checkParent = parent:GetParent()
+	end
+
+	return parent
+end
+
+---
+-- @ignore
+function PANEL:Init()
+	-- Make navArea scrollable
+	local navAreaScroll = vgui.Create("DScrollPanelTTT2", self)
+	navAreaScroll:SetVerticalScrollbarEnabled(true)
+	navAreaScroll:Dock(BOTTOM)
+	self.navAreaScroll = navAreaScroll
+
+	-- Split nav area into a grid layout
+	local navAreaScrollGrid = vgui.Create("DIconLayout", self.navAreaScroll)
+	navAreaScrollGrid:Dock(FILL)
+	self.navAreaScrollGrid = navAreaScrollGrid
+
+	-- Get the frame to be able to enable keyboardinput on searchbar focus
+	self.frame = getHighestParent(self)
+
+	-- This turns off the engine drawing
+	self:SetPaintBackgroundEnabled(false)
+	self:SetPaintBorderEnabled(false)
+	self:SetPaintBackground(false)
+end
+
+---
+-- This function sets the size of the searchBar.
+-- @param number widthBar
+-- @param number heightBar
+-- @realm client
+function PANEL:SetSearchBarSize(widthBar, heightBar)
+	if not self.searchBar then return end
+
+	self.searchBar:SetSize(widthBar, heightBar)
+end
+
+---
+-- This function enables or disables the searchBar.
+-- @param bool active
+-- @realm client
+function PANEL:EnableSearchBar(active)
+	if not active then
+		if self.searchBar then self.searchBar:Clear() end
+		return
+	end
+
+	-- Add searchbar on top	
+	local searchBar = vgui.Create("DSearchBarTTT2", self)
+	searchBar:SetUpdateOnType(true)
+	searchBar:SetPos(0, heightNavHeader)
+	searchBar:SetHeightMult(1)
+
+	searchBar.OnValueChange = function(slf, searchText)
+		self:GenerateSubmenuList(self.basemenuClass:GetMatchingSubmenus(searchText))
+	end
+
+	searchBar.OnGetFocus = function(slf)
+		self.frame:SetKeyboardInputEnabled(true)
+	end
+
+	searchBar.OnLoseFocus = function(slf)
+		self.frame:SetKeyboardInputEnabled(false)
+	end
+
+	self.searchBar = searchBar
+end
+
+---
+-- This function adds a button for a submenu in the list.
+-- @param menuClass submenuClass
+-- @return panel
+-- @realm client
+function PANEL:AddSubmenuButton(submenuClass)
+	local settingsButton = self.navAreaScrollGrid:Add("DSubmenuButtonTTT2")
+
+	settingsButton:SetTitle(submenuClass.title or submenuClass.type)
+	settingsButton:SetIcon(submenuClass.icon, submenuClass.iconFullSize)
+
+	settingsButton.PerformLayout = function(panel)
+		local width = panel:GetParent():GetSize()
+		panel:SetSize(width, heightNavButton)
```

Comment:
```suggestion
		panel:SetSize(panel:GetParent():GetWide(), heightNavButton)
```

I *think* this should work as well